### PR TITLE
Postgres Replication + Builds in Travis

### PR DESCRIPTION
**Pull request by krallin**
A few key changes here:

- This will build in Travis, so it includes the same scaffolding other repos to be built in Travis include.
- Made some tweaks to `run-database.sh`:
  - `--client` now forwards arguments to `psql`, so you can use non-interactively. 
  - certs are generated at startup; they're no longer baked into the image.
  - A lot more things are confiruable via the environment so tests can use them.
  - Uses `sudo` over `su -c` for better `set -o xtrace`.
- There's an external `test.sh` script that exercises replication.

I incorporated the changes from the original replication PR (#19). Only question I have is whether we should consider enabling `hot_standby` on followers only.

Cheers,

### CODE REVIEW

**Comment by krallin** [comment]
cc @fancyremarker

**Comment by fancyremarker** [comment]
> Only question I have is whether we should consider enabling hot_standby on followers only.

I believe this should be safe to set on both leader and follower(s), and doing so seems to simplify configuration. Are there any downsides to setting it on the leader?

File: bin/run-database.sh [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  sed -i "s:__DATA_DIRECTORY__:${DATA_DIRECTORY}:g" "$PG_CONF"
+  sed -i "s:__CONF_DIRECTORY__:${CONF_DIRECTORY}:g" "$PG_CONF"
+  sed -i "s:__RUN_DIRECTORY__:${RUN_DIRECTORY:-"$DEFAULT_RUN_DIRECTORY"}:g" "$PG_CONF"
+  sed -i "s:__PORT__:${PORT:-"$DEFAULT_PORT"}:g" "$PG_CONF"
+  sed -i "s:__PG_VERSION__:${PG_VERSION}:g" "$PG_CONF"
+
+  # Set up a self-signed SSL cert
+  mkdir -p "${CONF_DIRECTORY}/ssl"
+  cd "${CONF_DIRECTORY}/ssl"
+  openssl req -new -newkey rsa:1024 -days 365000 -nodes -x509 -keyout server.key -subj "/CN=PostgreSQL" -out server.crt
```
Comment (fancyremarker) [inline bin/run-database.sh]: Thanks for this change.

File: test.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
```
Comment (fancyremarker) [inline test.sh]: I didn't bring this up on other PRs, but is the idea behind using a plain old shell script for post-build tests that Bats is overkill? (Or that removing the Bats dependency simplifies the requirements for building/pushing the image?) A shell file like this one seems well suited when there is just one (or a couple) test example, but if we were to add more post-build tests, I could see how Bats could become useful again.

**Comment by fancyremarker** [comment]
:clap: Looks great to me, thanks. Please merge at will; none of my questions/comments are blockers. I see that there's one small change to make to .travis.yml before merge.

File: test.sh (hunk 2 above)
Comment (krallin) [inline test.sh]: I agree with you on all counts here; there aren't many tests (yet) being run outside of the container, so a shell script with `-o errexit` felt like the right option. However, if we start adding more, then using bats here as well would be a good idea.

It'd be a simple matter of installing bats in the `.travis.yml` and updating it across images that use it, and changing the shebang line! :smile:

Cheers,

**Pull request merged**